### DISCUSSION TITLE
Grafana 9 via Teleport Application

### DISCUSSION THREAD

**Discussion by qwerp** [gravitational/teleport#15375] [Q&A] [answered]
Hi, on the recent update to grafana 9, Teleport 10 via Application does not support the reverse Proxy of Websocket.

Most of the other basic functionality works on, however the "Live" trailing feature which depends on Websocket now nolonger works.

This is due to an new security feature put inplace to secure the Websocket.

Note: The same setup still works for Grafana 8.3.4 and below.
https://grafana.com/docs/grafana/latest/release-notes/release-notes-8-3-5/

as per official grafana documentation.
https://grafana.com/tutorials/run-grafana-behind-a-proxy/

which requires below config for NGINX.

`  # Proxy Grafana Live WebSocket connections.
  location /api/live/ {
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
    proxy_set_header Host $http_host;
    proxy_pass http://grafana;
  }`

Is there any way in teleport configuration that we are able to per have this included?

Thanks

**Comment by webvictim** [comment]
@r0mant I thought we made some changes to app access recently to make sure websockets have the same headers/config applied to them as regular connections?

**Comment by qwerp** [comment, marked as answer]
any recommendation what the "teleport.yaml" should look like to support that "changes"?

**Reply by webvictim** [reply]
There shouldn't need to be any changes made to support it. I know several folks are having issues with app access following Grafana's CSRF changes, however. I'm going to try and get these raised in an issue so they can be investigated.

**Reply by ChadOhman** [reply]
Whatever came of this? I can't find any documentation on using Grafana 9 with Teleport, even the training materials still use Grafana 8.
Thanks...

**Reply by webvictim** [reply]
Try something like this:

```
apps:
- name: grafana
  uri: http://localhost:3000
  rewrite:
    headers:
    - "Host: grafana.teleport.example.com"
    - "Origin: https://grafana.teleport.example.com"
```
